packages/utils/telemetry-utils/src/layerCompatError.ts

```diff
@@ -0,0 +1,64 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import {
+	checkLayerCompatibility,
+	type FluidLayer,
+	type ILayerCompatDetails,
+	type ILayerCompatSupportRequirements,
+} from "@fluid-internal/client-utils";
+import type { IErrorBase, ITelemetryBaseLogger } from "@fluidframework/core-interfaces";
+
+import { UsageError } from "./error.js";
+
+/**
+ * Validates the compatibility between two layers using their compatibility details and support requirements.
+ * @internal
+ */
+export function validateLayerCompatibility(
+	layer1: FluidLayer,
+	layer2: FluidLayer,
+	compatDetailsLayer1: Pick<ILayerCompatDetails, "pkgVersion" | "generation">,
+	compatSupportRequirementsLayer1: ILayerCompatSupportRequirements,
+	maybeCompatDetailsLayer2: ILayerCompatDetails | undefined,
+	disposeFn: (error?: IErrorBase) => void,
+	logger: ITelemetryBaseLogger,
```

Comment:
I don't think this should be a Base logger, but rather `ITelemetryLoggerExt` or something.  My other comment below probably depends on it being the internal one and calling `sendErrorEvent` to get the automatic pulling of `error.getTelemetryProperties` or whatever